src/uu/touch/src/touch.rs

```diff
@@ -236,96 +261,128 @@ pub fn uu_app() -> Command {
         )
 }
 
-// Determine the access and modification time
-fn determine_times(matches: &ArgMatches) -> UResult<(FileTime, FileTime)> {
-    match (
-        matches.get_one::<OsString>(options::sources::REFERENCE),
-        matches.get_one::<String>(options::sources::DATE),
-    ) {
-        (Some(reference), Some(date)) => {
-            let (atime, mtime) = stat(Path::new(&reference), !matches.get_flag(options::NO_DEREF))?;
-            let atime = filetime_to_datetime(&atime).ok_or_else(|| {
-                USimpleError::new(1, "Could not process the reference access time")
-            })?;
-            let mtime = filetime_to_datetime(&mtime).ok_or_else(|| {
-                USimpleError::new(1, "Could not process the reference modification time")
-            })?;
-            Ok((parse_date(atime, date)?, parse_date(mtime, date)?))
-        }
-        (Some(reference), None) => {
-            stat(Path::new(&reference), !matches.get_flag(options::NO_DEREF))
+/// Determine the access and modification time from a reference file
+fn determine_times_from_ref(
+    reference: &Path,
+    date: Option<&String>,
+    no_deref: bool,
+) -> UResult<(FileTime, FileTime)> {
+    if let Some(date) = date {
+        let (atime, mtime) = stat(reference, !no_deref)?;
+        let atime = filetime_to_datetime(&atime)
+            .ok_or_else(|| USimpleError::new(1, "Could not process the reference access time"))?;
+        let mtime = filetime_to_datetime(&mtime).ok_or_else(|| {
+            USimpleError::new(1, "Could not process the reference modification time")
+        })?;
+        Ok((parse_date(atime, date)?, parse_date(mtime, date)?))
+    } else {
+        stat(reference, !no_deref)
+    }
+}
+
+/// Execute the touch command.
+///
+/// # Errors
+///
+/// If the user doesn't have permission to access the file, or if one of the directory
+/// components of the file path doesn't exist
+pub fn touch(file: &InputFile, opts: &Options) -> UResult<()> {
+    // FIXME: find a way to avoid having to clone the path
+    // TODO (ysthakur) ^ Fixed with the Cow?
```

Comment:
Yes, we can remove these comments now :)

We have no need for entire conversations in the code.